include/ofi_rbuf.h

```diff
@@ -116,7 +116,11 @@ struct ofi_ringbuf {
 
 static inline int ofi_rbinit(struct ofi_ringbuf *rb, size_t size)
 {
+	if (size == 0 || size > (SIZE_MAX / 2))
+		return -EINVAL;
 	rb->size = roundup_power_of_two(size);
+	if (rb->size == 0)
+		return -EINVAL;
```

Comment:
This check seems to be redundant.

Reply:
Ok, thanks for catching that. Removed from recent commit.